src/main/java/hudson/plugins/emailext/plugins/content/FailedTestsContent.java

```diff
@@ -106,43 +98,125 @@ else if (result.getRun() != null) {
         }
     }
 
-    private int outputTest(StringBuilder buffer, TestResult failedTest,
-            boolean showStack, boolean showMessage, int lengthLeft) {
-        StringBuilder local = new StringBuilder();
-        String lineBreak = getLineBreak();
+    private SummarizedTestResult prepareSummarizedTestResult(AbstractTestResultAction<?> testResult) {
+        if(null == testResult) {
+            SummarizedTestResult result = new SummarizedTestResult(0, getLineBreak());
+            result.summary = "No tests ran.";
+            return result;
+        }
+        int failCount = testResult.getFailCount();
+        SummarizedTestResult result = new SummarizedTestResult(failCount, getLineBreak());
+        if(failCount == 0) {
+            result.summary = "All tests passed";
+        }
+        else {
+            result.summary = String.format("%d tests failed.", failCount);
+            setMaxLength();
+            if(maxTests > 0) {
+                int printSize = 0;
+                for (TestResult failedTest : testResult.getFailedTests()) {
+                    if (regressionFilter(failedTest)) {
+                        String stackTrace = showStack ? (escapeHtml ? escapeHtml(failedTest.getErrorStackTrace()) : failedTest.getErrorStackTrace()) : null;
+                        String errorDetails = showMessage ? (escapeHtml ? escapeHtml(failedTest.getErrorDetails()) : failedTest.getErrorDetails()) : null;
+                        String name = String.format("%s.%s", (failedTest instanceof CaseResult) ? ((CaseResult) failedTest).getClassName() : failedTest.getFullName(),
+                                failedTest.getDisplayName());
+                        FailedTest t = new FailedTest(name, failedTest.isPassed(), errorDetails, stackTrace);
+                        String testYaml = t.toString();
+                        if (printSize <= maxLength && result.tests.size() < maxTests) {
+                            result.tests.add(t);

```

Comment:
nit: If you decide to take my advise on the approximation of the line count, the `FailedTest` object will no longer be necessary here and it would then be cleaner to have an `addTest()` method in `SummarizedTestResult` to tuck away the above line and the one at 123.

Reply:
Not having a `FailedTest` object will result in a bad Yaml output as I am currently leveraging the yaml library to convert a `List<FailedTest>` to a yaml list structure. The `toString()` can be avoided by just adding up the string lengths of `stackTrace, errorDetails and name` but since we have some boilerplate text along with it in the actual output, i tried to mimic the same behavior in the `toString()` of `FailedTest` so that the estimate is approximately same for each `FailedTest`

Reply:
I am not suggesting to get rid off the `FailedTest` class or the use of it in the `SummarizedTestResult` datastructure. I am just referring to the construction of the object here as opposed to abstracting it into a new `addTest` method.